### ISSUE TITLE
Styx: Master of Shadows (242640)

### ISSUE THREAD

**Issue by djhi** [ValveSoftware/Proton#810]
Works flawlessly after installing `xact` with `winetricks`

**Comment by djhi** [comment]
There is one issue though: it crashes on exit and keeps multiple defunct processes

**Comment by jtux270** [comment]
Doesn't even start on my system. Performs first time set-up and then nothing. Xact is installed via winetricks, Proton v 3.7
[Gist](https://gist.github.com/jtux270/9d9de6876289f9f2581296689e65e0cc)

**Comment by lucifertdark** [comment]
It also needs dotnet40 which doesn't install at all in Proton.

**Comment by minisori** [comment]
I was able to install dotnet40 with latest winetricks and protontricks and got it working perfectly.

**Comment by lucifertdark** [comment]
> I was able to install dotnet40 with latest winetricks and protontricks and got it working perfectly. It does not need xact as it's already installed.

You sure you're using Proton? it's 64bit wine 3.7 & dotnet40 has been uninstallable for everyone.

**Comment by minisori** [comment]
Yes, i'm the first surprissed because i was never able to install it in a wine64 prefix before.
Maybe something in the new winetricks fixed it or who knows. I only had to reinstall dotnet40 with protontricks and it was working. I have been playing a while since then.

Winetricks from debian sid repositories:
winetricks -V
20180603 - sha256sum: 09d153c320ba4a58cfe600beab42913803ab0871b41ffea79c5b3bf6f0214a36

I'm not sure if xact is needed but i installed these two: xact dotnet40.

**Comment by jtux270** [comment]
> i'm the first surprissed because i was never able to install it in a wine64 prefix before.

Hmm, doesn't work on Linux Mint 18.2 

```
protontricks 242640 dotnet40 xact
[INFO] Found Steam directory at /home/mint/.steam/steam. You can also define Steam directory manually using $STEAM_DIR
[INFO] WINETRICKS environment variable is not available. Falling back to /usr/bin/winetricks
[INFO] Found proton version 3.7. You can also define the proton version manually using $PROTON_VERSION
[INFO] WINE environment variable is not available Setting WINE environment variable to proton bundled version
[INFO] WINESERVER environment variable is not available Setting WINESERVER environment variable to proton bundled version
[INFO] Found 0 Steam library folders
[INFO] Found the prefix directory at /home/mint/.steam/steam/steamapps/compatdata/242640/pfx
------------------------------------------------------
You are using a 64-bit WINEPREFIX. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing w_do_call dotnet40
Executing load_dotnet40
------------------------------------------------------
This package does not work on a 64-bit installation
------------------------------------------------------
```

```
winetricks -V
20180815-next - sha256sum: da6c591669282bed53aa8c260d496ac30bb1fe30341e967ec98f03688777d312
```

Are we missing something here?

**Comment by minisori** [comment]
That's odd, mine does install. I've installed it in another game to get the log and not mess with this one because i'm playing it.

The installation log:
https://gist.github.com/minisori/95b730920cc61c85f7303d72d48f381e

Steam info:
https://gist.github.com/minisori/217be39a24ae8a58e7d70a022df8c481

Winetricks used:
https://packages.debian.org/sid/winetricks

Edit:
A little more investigation, when the game is launched i get some dotnet errors in a window, they are gone after reinstalling it with --force dotnet40 and repairing the installation.

**Comment by John-Gee** [comment]
I wasn't able to install dotnet with standard wine/winetricks, but using protontricks did work for me as it did for minisori and I have latest winetricks like jtux270.

**Comment by lucifertdark** [comment]
Just installing a small game that needs dotnet40 to test but could it be something as simple as using --force in the command to get dotnet to install? will report back shortly if it works with that for me.

edit: well I removed winetricks & replaced it with the exact same version, installed ROTT & did what you did & nothing, we are definitely missing something somewhere that's different on your system.

ps I'm using protontricks as well, not just winetricks on it's own.

**Comment by John-Gee** [comment]
I didn't use force, full command:

STEAM_DIR=<PATH>/SteamLibrary protontricks 242640 dotnet40 xact

I'd like to know why using my system's wine with winetrick didn't work, but I have no clue. Styx' installer installed dotnet, and most likely failed, and then the command above repaired the installation (+ xact).

**Comment by minisori** [comment]
> I didn't use force, full command:
> 
> STEAM_DIR=/SteamLibrary protontricks 242640 dotnet40 xact
> 
> I'd like to know why using my system's wine with winetrick didn't work, but I have no clue. Styx' installer installed dotnet, and most likely failed, and then the command above repaired the installation (+ xact).

To use standard winetricks with proton you need to set some enviroment variables so the same wine and winserver are use:

export WINESERVER="$STEAM_DIR/steamapps/common/Proton\ 3.7/dist/bin/wineserver"
export WINELOADER="$STEAM_DIR/steamapps/common/Proton\ 3.7/dist/bin/wine"
WINEPREFIX="$STEAM_DIR/steamapps/compatdata/242640/pfx" winetricks dotnet40 xact


The point of protontricks it's to set all these variables for you.

**Comment by John-Gee** [comment]
That's not what I meant.

I set the prefix and used my distro's wine and winetricks, as if it was any other Wine setup, and it failed to correctly install dotnet, I tried 40, 45, and 46 and they all failed.

I'm guessing my current Wine (staging 3.16) has a problem there, or is simply incompatible with latest winetricks.

**Comment by lucifertdark** [comment]
The weirdest thing happened, it worked for me this time too, I now have a working copy of Master of Shadows, the only thing I did differently to what I normally do is run protontricks from inside the 242640 folder itself protontricks 242640 dotnet40 xact, I did the same thing John-Gee did chose repair dotnet40 & it just worked. Very strange but encouraging, if it worked for this one it "should" work for other games the same way.

**Comment by minisori** [comment]
Yes, it looks like you have to install it twice, the second one with --force and do the repair.

**Comment by John-Gee** [comment]
I did not run it from the 242640 folder, nor use the --force option, so I doubt either of these are what made it work :)

**Comment by lucifertdark** [comment]
> > I didn't use force, full command:
> > STEAM_DIR=/SteamLibrary protontricks 242640 dotnet40 xact
> > I'd like to know why using my system's wine with winetrick didn't work, but I have no clue. Styx' installer installed dotnet, and most likely failed, and then the command above repaired the installation (+ xact).
> 
> To use standard winetricks with proton you need to set some enviroment variables so the same wine and winserver are use:
> 
> export WINESERVER="$STEAM_DIR/steamapps/common/Proton\ 3.7/dist/bin/wineserver"
> export WINELOADER="$STEAM_DIR/steamapps/common/Proton\ 3.7/dist/bin/wine"
> WINEPREFIX="$STEAM_DIR/steamapps/compatdata/242640/pfx" winetricks dotnet40 xact
> 
> The point of protontricks it's to set all these variables for you.

Now I'm totally confused, those commands are doing nothing for me, every time I use winetricks with them (even edited to show the exact path" I'm getting the 3.16 wine instead, this is giving me a headache.

**Comment by minisori** [comment]
Try with WINE instead of WINELOADER.

**Comment by simons-public** [comment]
I hope this isn't the wrong place for this, but my project [protonfixes](https://github.com/simons-public/protonfixes) has a builtin fix for Styx:

**Comment by jtux270** [comment]
> I hope this isn't the wrong place for this, but my project [protonfixes](https://github.com/simons-public/protonfixes) has a builtin fix for Styx:

May I have your help in dealing with protonfixes? I (seemingly) intalled it following the instructions but Styx is still a no go :( Do you have a bugzilla or any other means of discussion somewhere? My system is Mint 18.2

**Comment by simons-public** [comment]
Sure, it's [here on Github](https://github.com/simons-public/protonfixes)

Nevermind, just realized you found it :)

**Comment by madewokherd** [comment]
Works for me in 64-bit with wine-mono, not sure why I haven't needed xact.

**Comment by NoXPhasma** [comment]
The game works out of the box for me with Proton 5.0-9. No issues with audio or dotnet.

Sysinfo: https://gist.github.com/NoXPhasma/0ff4dc01d36e1f47800b2b346a31630c

**Comment by gordan-ramsey** [comment]
The game works out of the box, however mouse won't lock.
If i use the OG steam controller it works just fine.

Using --force-grab-cursor fixes the issue for the mouse.

it's exactly the same problem that i've reported in Mirror's Edge https://github.com/ValveSoftware/Proton/issues/355#issuecomment-3857729334

I've tested the game with Proton Hotfix, 10 and 11 beta.

I have set Highest Graphics settings. I get on avg 60 FPS, lowest is 31 and highest is 114.

Hardware:
3 Monitors (Samung S34E790C, Samsung S32D850T, ASUS VE278) all connected through displayport
CPU: AMD Ryzen 7 3700x
MB: ASUS X570-l
Storage: 2x 2TB M.2 SSDs Sk hynix Platinum P41 SSD PCI-e 4.0.
RAM: 64GB DDR4 @ 3000Mhz
GPU: Nvidia RTX 3060 (Driver Version: 580.159.03 CUDA Version: 13.0)
Mouse: wireless LogiTech M170
OS: Pop! 24.04
Kernel: `Linux pop-os 6.18.7-76061807-generic #202601231045~1777506123~24.04~a891078 SMP PREEMPT_DYNAMIC Thu A x86_64 x86_64 x86_64 GNU/Linux`

<img width="1001" height="585" alt="Image" src="https://github.com/user-attachments/assets/04f936b8-5af2-4409-8eb6-c14c79ef7dab" />
